File: Excel/695 Challenge.py
import pandas as pd
from datetime import datetime, timedelta

def monday_of_isoweek(year, week):
    return datetime(year, 1, 4) + timedelta(days=(week - 1) * 7 - (datetime(year, 1, 4).weekday()))

data = [{
    "From": monday_of_isoweek(2025, week),
    "To": monday_of_isoweek(2025, week) + timedelta(days=6),
    "Week": week
} for week in range(1, 53)]

df = pd.DataFrame(data)
df['Month'] = df['To'].dt.strftime('%b')
df['Month Week'] = df.groupby('Month').cumcount() + 1
df['From'] = df['From'].dt.strftime('%Y-%m-%d')
df['To'] = df['To'].dt.strftime('%Y-%m-%d')

result = df[['Month', 'Month Week', 'Week', 'From', 'To']].rename(columns={
    'Month Week': 'Month Week',
    'Week': 'Year Week',
    'From': 'From Date',
    'To': 'To Date'
})

print(result)

File: Excel/600-699/695/695 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/695 Broadcast Calendar.xlsx"
test = read_excel(path, range = "A2:E54")

monday_of_isoweek <- function(year, week) {
  ISOdate(year, 1, 4) +
    weeks(week - 1) -
    days(wday(ISOdate(year, 1, 4), week_start = 1) - 1)
}

result = data.frame(
  From = monday_of_isoweek(2025, 1:52),
  To = monday_of_isoweek(2025, 1:52) + days(6),
  Week = 1:52
) %>%
  mutate(Month = month(To, label = TRUE, abbr = TRUE, locale = "en")) %>%
  mutate(month_week = row_number(), .by = Month) %>%
  mutate(From = format(From, "%Y-%m-%d"), To = format(To, "%Y-%m-%d")) %>%
  select(
    Month,
    `Month Week` = month_week,
    `Year Week` = Week,
    `From Date` = From,
    `To Date` = To
  )
